lib/console/ai/evidence/component/persistent_volume_claim.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Console.AI.Evidence.Component.PersistentVolumeClaim do
+  use Console.AI.Evidence.Base
+  alias Kazan.Apis.Storage.V1, as: StorageV1
+  alias Kazan.Apis.Core.V1, as: CoreV1
+  alias Console.Repo
+  alias Console.AI.Evidence.Logs
+  alias Console.Schema.{Cluster, OperationalLayout}
+
+  def hydrate(%CoreV1.PersistentVolumeClaim{} = pvc) do
+    messages = [
+      {:user, "the persistent volume claim #{component(pvc)} has a current state of:\n#{json_blob(encode(pvc))}"}
+    ] ++ pv_message(pvc)
+
+    messages
+    |> maybe_add_csi_logs(pvc, Repo.preload(get_cluster(), [:operational_layout]))
+  end
+  def hydrate(_), do: {:ok, []}
+
+  defp pv_message(%CoreV1.PersistentVolumeClaim{spec: %{volume_name: name}}) when is_binary(name) do
+    with {:ok, pv} <- CoreV1.read_persistent_volume!(name) |> Kube.Utils.run() do
+      [
+        {:user, "the claim is bound to the persistent volume #{name}, which has a state of:\n#{json_blob(encode(pv))}"}
+      ]
+    else
+      _ -> []
+    end
+  end
+  defp pv_message(_), do: []
+
+  defp maybe_add_csi_logs(history, %CoreV1.PersistentVolumeClaim{spec: %{storage_class_name: sc_name}}, %Cluster{
+    operational_layout: %OperationalLayout{namespaces: %OperationalLayout.Namespaces{} = ns}
+  } = cluster) when is_binary(sc_name) do
+    with {:ok, %StorageV1.StorageClass{provisioner: p}} <- StorageV1.read_storage_class!(sc_name) |> Kube.Utils.run(),
```

Comment:
small stylistic tweak, but I think this might be more readable if you just pipe into case eg:

```elixir
StorageV1.read_storage_class!(sc_name)
|> Kube.Utils.run()
|> case do
  {:ok, %StorageV1.StorageClass{provisioner: p}} ->
     ...
  _ -> {:ok, history}
end
```

(generally with clauses are useful if you have > 1 sequential match or don't use the else clause)